imdraw/imdraw.go

```diff
@@ -357,15 +357,25 @@ func (imd *IMDraw) fillPolygon() {
 	}
 
 	off := imd.tri.Len()
-	imd.tri.SetLen(imd.tri.Len() + 3*(len(points)-2))
-
-	for i, j := 1, off; i+1 < len(points); i, j = i+1, j+3 {
-		for k, p := range []point{points[0], points[i], points[i+1]} {
-			(*imd.tri)[j+k].Position = p.pos
-			(*imd.tri)[j+k].Color = p.col
-			(*imd.tri)[j+k].Picture = p.pic
-			(*imd.tri)[j+k].Intensity = p.in
-		}
+	imd.tri.SetLen(off + 3*(len(points)-2))
+
+	for i, j := 1, off; i+1 < len(points); {
+		(*imd.tri)[j].Position = points[0].pos
+		(*imd.tri)[j].Color = points[0].col
+		(*imd.tri)[j].Picture = points[0].pic
+		(*imd.tri)[j].Intensity = points[0].in
+		j++
+		(*imd.tri)[j].Position = points[i].pos
+		(*imd.tri)[j].Color = points[i].col
+		(*imd.tri)[j].Picture = points[i].pic
+		(*imd.tri)[j].Intensity = points[i].in
+		j++
+		i++
+		(*imd.tri)[j].Position = points[i].pos
+		(*imd.tri)[j].Color = points[i].col
+		(*imd.tri)[j].Picture = points[i].pic
+		(*imd.tri)[j].Intensity = points[i].in
+		j++
```

Comment:
This is ugly. The reason why the former version was slow could be just because of the use of slice. Try replacing it with a static array `[...]point{` instead of `[]point{` and see if it makes a difference.